src/pages/host/list.jsx

```diff
@@ -1,16 +1,46 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import "../../server";
 
 function List() {
-  return (
+  const [list, setList] = useState([]);
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    const fecthList = async () => {
+      try {
+        const res = await fetch("/api/vans");
+        const listData = await res.json();
+        setList(listData.vans);
+        setLoading(false);
+      } catch (err) {
+        setList("This is an error");
+        setLoading(false);
+      }
+    };
+    fecthList();
+    return () => {
+      fecthList();
+    };
+  }, []);
+  return loading ? (
+    <div className="text-center py-52">Loading...</div>
+  ) : (
     <div className="bg-[#fff7ed] px-4 py-8 ">
-      <div className="flex justify-between">
-        <p className="font-bold">Your listed vans</p>
-        <button>View all</button>
-      </div>
-      <div className="py-12">
-        <p className="text-center text-red-800 font-bold">
-          You haven't rented any van yet!
-        </p>
+      <h2 className="font-bold text-4xl">Your listed vans</h2>
+      <div className="py-12 flex flex-col gap-4">
+        {list.map((item, idx) => (
+          <Link
+            to={`/host/list/${idx}`}
```

Comment:
Rather than using the item index, use the item id, as the index might not match the id it has.